mixer/adapter/redisquota/redisquota_integration_test.go

```diff
@@ -107,149 +107,93 @@ spec:
-		attrs      map[string]interface{}
-		quotas     map[string]istio_mixer_v1.CheckRequest_QuotaParams
-		statusCode int32
-		expected   map[string]int64
-	}{
-		"Request 30 when 50 is available": {
-			attrs: map[string]interface{}{},
-			quotas: map[string]istio_mixer_v1.CheckRequest_QuotaParams{
-				"key1": {
-					Amount:     30,
-					BestEffort: true,
-				},
-			},
-			expected: map[string]int64{
-				"key1": 30,
-			},
-			statusCode: 0,
-		},
-		"Exceed allocation request with bestEffort": {
-			attrs: map[string]interface{}{},
-			quotas: map[string]istio_mixer_v1.CheckRequest_QuotaParams{
-				"key2": {
-					Amount:     60,
-					BestEffort: true,
-				},
-			},
-			expected: map[string]int64{
-				"key2": 50,
-			},
-			statusCode: 0,
-		},
-		"Exceed allocation request without bestEffort": {
-			attrs: map[string]interface{}{},
-			quotas: map[string]istio_mixer_v1.CheckRequest_QuotaParams{
-				"key3": {
-					Amount:     60,
-					BestEffort: false,
-				},
-			},
-			expected: map[string]int64{
-				"key3": 0,
-			},
-			statusCode: 0,
-		},
-		"Dimension override with best effort": {
-			attrs: map[string]interface{}{
-				"source.service":      "reviews",
-				"destination.service": "ratings",
-			},
-			quotas: map[string]istio_mixer_v1.CheckRequest_QuotaParams{
-				"overridden": {
-					Amount:     15,
-					BestEffort: true,
-				},
-			},
-			expected: map[string]int64{
-				"overridden": 12,
-			},
-			statusCode: 0,
-		},
+func TestFixedWindowAlgorithm(t *testing.T) {
+	// start mock redis server
+	mockRedis, err := miniredis.Run()
+	if err != nil {
+		t.Fatalf("Unable to start mock redis server: %v", err)
 	}
+	defer mockRedis.Close()
 
-	for id, c := range cases {
-		// start mock redis server
-		mockRedis, err := miniredis.Run()
-		if err != nil {
-			t.Fatalf("Unable to start mock redis server: %v", err)
-		}
-		defer mockRedis.Close()
+	// start mixer with redisquota adapter
+	args := server.NewArgs()
 
-		// start mixer with redisquota adapter
-		args := server.NewArgs()
+	args.APIPort = 0
+	args.MonitoringPort = 0
+	args.Templates = template.SupportedTmplInfo
+	args.Adapters = []adapter.InfoFn{
+		GetInfo,
+	}
 
-		args.APIPort = 0
-		args.MonitoringPort = 0
-		args.Templates = template.SupportedTmplInfo
-		args.Adapters = []adapter.InfoFn{
-			GetInfo,
-		}
+	serviceCfg := adapterConfig
+	serviceCfg = strings.Replace(serviceCfg, "__REDIS_SERVER_ADDRESS__", mockRedis.Addr(), -1)
+	var cerr error
+	if args.ConfigStore, cerr = storetest.SetupStoreForTest(globalConfig, serviceCfg); cerr != nil {
+		t.Fatal(cerr)
+	}
 
-		serviceCfg := adapterConfig
-		serviceCfg = strings.Replace(serviceCfg, "__RATE_LIMIT_ALGORITHM__", algorithm, -1)
-		serviceCfg = strings.Replace(serviceCfg, "__REDIS_SERVER_ADDRESS__", mockRedis.Addr(), -1)
-		var cerr error
-		if args.ConfigStore, cerr = storetest.SetupStoreForTest(globalConfig, serviceCfg); cerr != nil {
-			t.Fatal(cerr)
-		}
+	mixerServer, err := server.New(args)
```

Comment:
alternative may be start the mixer once, and run all testcases as requests?

Reply:
It requires different configurations for rolling window and fixed window. 
I think for the integration test, this is good enough.